File: source/includes/data-formats/time-series.java
package org.example;

import com.mongodb.ConnectionString;
import com.mongodb.MongoClientSettings;

import com.mongodb.client.model.CreateCollectionOptions;
import com.mongodb.client.model.TimeSeriesOptions;
import com.mongodb.client.result.InsertManyResult;
import com.mongodb.reactivestreams.client.*;
import org.bson.Document;
import reactor.core.publisher.Flux;

import java.util.Arrays;
import java.util.Date;

public class Main {
    public static void main(String[] args) {
        
        // Replace the placeholder with your Atlas connection string
        String uri = "<connection string URI>";

        MongoClientSettings settings = MongoClientSettings.builder()
                .applyConnectionString(new ConnectionString(uri))
                .build();

        // Create a new client and connect to the server
        try (MongoClient mongoClient = MongoClients.create(settings)) {
            // start-create-time-series
            MongoDatabase database = mongoClient.getDatabase("fall_weather");

            TimeSeriesOptions tsOptions = new TimeSeriesOptions("timestamp");
            CreateCollectionOptions collectionOptions = new CreateCollectionOptions().timeSeriesOptions(tsOptions);

            database.createCollection("october2024", collectionOptions);
            // end-create-time-series

            // start-print-time-series
            ListCollectionsPublisher<Document> listCollectionsPublisher = database.listCollections();

            Flux.from(listCollectionsPublisher)
                    .doOnNext(System.out::println)
                    .blockLast();
            // end-print-time-series

            // start-insert-time-series-data
            MongoCollection<Document> collection = database.getCollection("october2024");

            // Temperature data for October 1, 2024
            Document temperature1 = new Document("temperature", 54)
                    .append("location", "New York City")
                    .append("timestamp", new Date(1727755200000L));

            // Temperature data for October 2, 2024
            Document temperature2 = new Document("temperature", 55)
                    .append("location", "New York City")
                    .append("timestamp", new Date(1727841600000L));

            Publisher<InsertManyResult> insertPublisher =
                    collection.insertMany(Arrays.asList(temperature1, temperature2));
            Mono.from(insertPublisher).block();
            // end-insert-time-series-data
        }
    }
}

File: source/includes/data-formats/time-series.kt
package org.example
import com.mongodb.ConnectionString
import com.mongodb.MongoClientSettings
import com.mongodb.client.model.*
import com.mongodb.kotlin.client.MongoClient
import org.bson.json.JsonWriterSettings
import java.util.Date
import org.bson.Document

fun main() {
    val uri = "<connection string URI>"

    val settings = MongoClientSettings.builder()
        .applyConnectionString(ConnectionString(uri))
        .retryWrites(true)
        .build()

    val mongoClient = MongoClient.create(settings)

    // start-create-time-series
    val database = mongoClient.getDatabase("fall_weather")

    val tsOptions = TimeSeriesOptions("timestamp")
    val collectionOptions = CreateCollectionOptions().timeSeriesOptions(tsOptions)

    database.createCollection("october2024", collectionOptions)
    // end-create-time-series

    // start-print-time-series
    val results = database.listCollections()
    val jsonSettings = JsonWriterSettings.builder().indent(true).build()

    results.forEach { result ->
        println(result.toJson(jsonSettings))
    }
    // end-print-time-series

    // start-insert-time-series-data
    val collection = database.getCollection<Document>("october2024")

    // Temperature data for October 1, 2024
    val temperature1 = Document("temperature", 54)
                        .append("location", "New York City")
                        .append("timestamp", Date(1727755200000))

    // Temperature data for October 2, 2024
    val temperature2 = Document("temperature", 55)
                        .append("location", "New York City")
                        .append("timestamp", Date(1727841600000))

    collection.insertMany(listOf(temperature1, temperature2))
    // end-insert-time-series-data
}

